### ISSUE TITLE
implement copy method/constructor for `ml::train::Model`

### ISSUE THREAD

**Issue by JRazek** [nnstreamer/nntrainer#1889]
Hello, I am trying to copy a `ml::train::Model` object that is a neural network. Unfortunately ccapi does not provide an explicit way to copy it. 

There is method like this implemented in nntrainer, but it is not possible to access through ccapi.
```cpp
nntrainer::NeuralNetwork::copy(nntrainer::NeuralNetwork&)
```
I would need something such as this:
```cpp
ml::train::Model::copy(ml::train::Model const&);
```

Please implement this feature.

**Comment by jijoongmoon** [comment]
Hello, Thanks for the request. 
We will let you know when it is ready.

**Comment by kparichay** [comment]
Meanwhile, the feature gets ready, you can consider an alternative way to create a copy of the model via saving the model as INI:
```c++
// ml::train::model m1 and m2 objects exist

// create the m1 model architecture with API/ini
m1.compile();
m1.initialize();
m1.save(FILE_LOC, ModelFormat::MODEL_FORMAT_INI); // save the model architecture to ini file at FILE_LOC

// load the same architecture for m2 model
m2.loadFromConfig(FILE_LOC);
// now m2 model has the same architecture as m1
```
You can extend this method with `ModelFormat::MODEL_FORMAT_INI_WITH_BIN` to include the weights as well.

@jijoongmoon can elaborate more on this if needed.

**Comment by jijoongmoon** [comment]
Thanks @kparichay !  As @kparichay mentioned, it is good alternative way you can use.

**Comment by jijoongmoon** [comment]
We provide the API to copy the neural network model configuration and getter/setter of Weights. #1893

**Comment by JRazek** [comment]
Feature was implemented with a #1893.
